src/JsonSchema/Validator.elm

```diff
@@ -89,6 +93,37 @@ validateMaxItems int values =
                 [ ( [], HasMoreItemsThan maxItems ) ]
 
 
+validateUniqueItems : Bool -> Maybe Schema -> Array.Array Value -> List Error
+validateUniqueItems unique item values =
+    if not unique then
+        []
+    else
+        case (Debug.log "items" item) of
+            Just ((String _) as itemSchema) ->
```

Comment:
I think see what you're doing here.

I think it might be easier to do the uniqueness calculation without making any expectations of the json value we're expecting. If you can decode your values into a JSON type like the one below, it's very easy to write an equality function for that to implement the uniqueness check.

```elm
-- A type that can hold any JSON value:
type Json
  = JsonString String
  | JsonNumber Float
  | JsonBoolean Boolean
  | Null
  | Array (List Json)
  | Object (Dict String Json)

equal : Json -> Json -> Bool
equal json1 json2 = 
  case (json1, json2) of
    (Object dict1, Object dict2) ->
      ...
```